packages/agent-toolkit/src/examples/mcp.ts

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env node
```

Comment:
This is great, though as you point it, it'd be nice if we just made this its own sub-package so folks can just use it not as an example, but rather as a standalone mcp implementation, which _also_ acts as an example!

Note also that this (https://github.com/wong2/litemcp/) is a pretty nice framework for building mcp services very easily.